core-api/src/main/java/com/optimizely/ab/config/audience/match/NumericAttributeCondition.java

```diff
@@ -0,0 +1,120 @@
+/**
+ *
+ *    Copyright 2019, Optimizely and contributors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+package com.optimizely.ab.config.audience.match;
+
+import com.optimizely.ab.internal.AttributesUtil;
+import com.optimizely.ab.internal.InvalidAudienceCondition;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import java.util.Objects;
+
+/**
+ * Abstract condition for {@link Number} comparisons.
+ * Comparisons are made by {@link Double#compare(double, double)}.
+ */
+public abstract class NumericAttributeCondition extends AbstractAttributeCondition<Number, Number> {
```

Comment:
It's interesting to have a class that handles multiple match types for a particular condition value type. Other SDKs don't really do this; they always branch first on the match type and then on the value type. Though this isn't a huge concern for me; like you, I'd rather figure out the class hierarchy and see what falls out of that.